### PR TITLE
fix: add /v1 prefix to the all paths and change 200 response of /generate path

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

As in title:
- add /v1 prefix to the all paths - as described in the `openapi.yaml`
- fix 200 response of /generate path: change `application/json` to the `application/octet-stream`.

I don't want to treat is a feat, but as fix, we forgot about that "things" in the previous PRs.

**Related issue(s)**
Fixes https://github.com/asyncapi/server-api/issues/34

### CODE REVIEW

**Comment by magicmatatjahu** [comment]
@fmvilas I had to fix tests :)

**Comment by fmvilas** [comment]
tests? what's that? :trollface:

**Review by fmvilas — DISMISSED** [review]

**Comment by magicmatatjahu** [comment]
@fmvilas I see that you have admin privileges to accept PR in this repo, but I would like to wait for review from @smoya or @BOLT04 😄

**Comment by fmvilas** [comment]
Yeah, I realized after approving. Please just take my approval as a 👍 from a contributor, not a code owner 🙏

**Review by smoya — APPROVED** [review]
LGTM!

**Comment by magicmatatjahu** [comment]
@smoya I fixed also in this PR problem with DockerFile that we know. Could you check again?

File: Dockerfile
Comment (smoya) [inline Dockerfile:18]: Can we create a follow up issue so we can fix this once https://github.com/asyncapi/.github/issues/123 is solved (I hope it does soon)? Removing the package-lock.json file here has unexpected behavior as we don't control which version is gonna be installed of each package if they are not pinned to an specific version...

Comment (magicmatatjahu) [reply Dockerfile:18]: I'm also against but sometimes switching from v1 to v2 (of packageLock version) can create errors, so bad and so not good 😆  but ok, we can go with that `package-lock.json`

Comment (magicmatatjahu) [reply Dockerfile]: @smoya changed :)

Comment (magicmatatjahu) [reply Dockerfile]: And also issue is created https://github.com/asyncapi/server-api/issues/36

Comment (smoya) [reply Dockerfile]: Oh no, it's not switching to one or another version something that scares me, no.
The fact we are not downloading exactly the same dependency versions as expected unless they are pinned to an exact version... this scares me 😆

Comment (magicmatatjahu) [reply Dockerfile]: @smoya So current changes are ok for you or we should hardcode version of the dependencies?

Comment (smoya) [reply Dockerfile]: In fact, I thing this change should not be added into this PR but to a new one so:

- We don't hide side effects. This change is totally unexpected according to the name of the final commit that will be merged.
- We can quickly and easily revert the changes as they will be isolated.

Comment (magicmatatjahu) [reply Dockerfile]: Reverted

Comment (magicmatatjahu) [reply Dockerfile]: @smoya Created new PR https://github.com/asyncapi/server-api/pull/37

**Review by smoya — APPROVED** [review]

File: openapi.yaml
Comment (BOLT04) [inline openapi.yaml:41]: what is the difference between `application/octet-stream` and `application/zip`? I'm asking because in my tests we return the `application/zip` Content-Type:
imagem

I believe we are sending `zip` because of this line: https://github.com/asyncapi/server-api/blob/03c5c53a351e54d07e5ff767ad25ee8295f6a800/src/services/archiver.service.ts#L19

Since the filename argument has a `.zip` extension, the header is set to that. But should we send just a binary file without specifying it's a zip to be more generic, what does everyone think?

Comment (smoya) [reply openapi.yaml:41]: Thats a really good catch @BOLT04 ! Then `application/zip` is more accurate. `application/octet-stream` would be valid as well to as `application/zip` inherits the features from `application/octet-stream` but it's not accurate.

As per IANA, it is an accepted MIME type since 1993 https://www.iana.org/assignments/media-types/application/zip

Comment (magicmatatjahu) [reply openapi.yaml]: Ok, changed to the `application/zip`. Thanks @BOLT04 for that catch! :)

**Review by smoya — APPROVED** [review]
LGTM!🚀

**Review by BOLT04 — APPROVED** [review]
LGTM 👍

**Pull request merged**
